app/services/builders/chartkick/base.rb

```diff
@@ -9,15 +9,43 @@ def initialize(entity_id, query)
       private
 
       def build_data(metrics)
-        metrics.each.inject({}) do |hash, metric|
+        metrics.inject({}) do |hash, metric|
           hash.merge!(
             metric.value_timestamp.strftime('%Y-%m-%d').to_s => value_in_hours_for(metric)
           )
         end
       end
 
-      def value_in_hours_for(metric)
-        (metric.value.to_f / 1.hour.seconds).round(2)
+      def build_distribution_data(entities)
+        entities.inject({}) do |hash, entity|
+          interval = resolve_interval(entity)
+          hash.merge!(
+            interval => value_in_hours_for(entity)
+          )
+        end
+      end
+
+      def value_in_hours_for(entity)
+        (entity.value.to_f / 1.hour.seconds).round(2)
+      end
+
+      def resolve_interval(entity)
+        entity_value = entity.value
+        if entity_value < 12
```

Comment:
Perhaps we can have a distributions hash, like:

```
{
  0..12 => '1-12',
  12..24 => '12-24'
...
}
```

And when you have a value, you check which of the hash keys contains your value. I don't know if it simplifies the code, but at least it will make rubocop and codeclimate stop complaining.

Reply:
```
                    user        system         total         real
without hash      0.001478     0.000003       0.001481    (0.001478)
with hash         0.064149     0.000294       0.064443    (0.064776)
```
The code for `with hash` 
```
INTERVALS.select { |interval, _| interval.include?(entity_value) }.values.first
```

Reply:
I'm fine with either approach 👍